examples/dpr_evaluation.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import torch
+import gc
+import numpy as np
+from tqdm import tqdm
+
+from farm.data_handler.processor import TextSimilarityProcessor
+from farm.data_handler.utils import read_dpr_json
+from farm.modeling.biadaptive_model import BiAdaptiveModel
+from farm.modeling.tokenization import Tokenizer
+from farm.utils import set_all_seeds, initialize_device_settings
+from farm.infer import Inferencer
+
+def get_unique_queries(dpr_data):
+    return [{'query': query} for query in set(map(lambda q: q['query'], dpr_data))]
+
+def get_unique_passages(dpr_data):
+    return list({passage['external_id']: passage for entry in dpr_data for passage in entry['passages']}.values())
+
+def average_rank(query_tensors, passage_tensors, dpr_data, batch_size = 128):
+    ranks = []
+    n_data = len(dpr_data)
+    docs_tensor = torch.transpose(passage_tensors, 0, 1).cuda()
+    n_batches = n_data // batch_size + 1 if n_data % batch_size > 0 else 0
+    for idx in tqdm(range(n_batches)):
+        queries_t = []
+        target_idxs = []
+        start_idx = idx * batch_size
+        end_idx = min([idx * batch_size + batch_size, n_data])
+        actual_batch_length = end_idx - start_idx
+        for d in dpr_data[start_idx:end_idx]:
+            query_idx = query_to_idx[d['query']]
+            query_tensor = query_tensors[query_idx]
+            queries_t.append(query_tensor)
+            target_id = list(filter(lambda x: x['label'] == 'positive', d))[0]['external_id']
+            target_idx = passage_to_idx[target_id]

```

Comment:
The same for `passage_to_idx`: did you mean to pass it as an argument as well otherwise it won't work?